packages/ui/components/form/select/SelectDownshift.tsx

```diff
@@ -0,0 +1,248 @@
+    });
+  }, [inputValue, props.items, props.searchableKey]);
+
+  const { getSelectedItemProps, getDropdownProps, removeSelectedItem } = useMultipleSelection({
+    selectedItems,
+    onStateChange({ selectedItems: newSelectedItems, type }) {
+      switch (type) {
+        case useMultipleSelection.stateChangeTypes.SelectedItemKeyDownBackspace:
+        case useMultipleSelection.stateChangeTypes.SelectedItemKeyDownDelete:
+        case useMultipleSelection.stateChangeTypes.DropdownKeyDownBackspace:
+        case useMultipleSelection.stateChangeTypes.FunctionRemoveSelectedItem:
+          if (newSelectedItems) {
+            setSelectedItems(newSelectedItems);
+            props.onSelectedItemsChange && props.onSelectedItemsChange(newSelectedItems);
+          }
+          break;
+        default:
+          break;
+      }
+    },
+  });
+  const {
+    isOpen,
+    getToggleButtonProps,
+    getLabelProps,
+    getMenuProps,
+    getInputProps,
+    getItemProps,
+    highlightedIndex,
+  } = useCombobox({
+    items,
+    stateReducer(state, actionAndChanges) {
+      const { changes, type } = actionAndChanges;
+
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.ItemClick:
+        case useCombobox.stateChangeTypes.InputBlur:
+          return {
+            ...changes,
+            ...(changes.selectedItem && { isOpen: true, highlightedIndex: changes.highlightedIndex }),
+          };
+        default:
+          return changes;
+      }
+    },
+    onStateChange({ type, selectedItem: newSelectedItem, inputValue: newInputValue }) {
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.ItemClick:
+          if (newSelectedItem) {
+            if (selectedItems.includes(newSelectedItem)) {
+              removeSelectedItem(newSelectedItem);
+              props.onSelectedItemsChange && props.onSelectedItemsChange(selectedItems);
+              return;
+            }
+            const newSelectedItems = [...selectedItems, newSelectedItem];
+            setSelectedItems(newSelectedItems);
+            props.onSelectedItemsChange && props.onSelectedItemsChange(newSelectedItems);
+          }
+          break;
+
+        case useCombobox.stateChangeTypes.InputChange:
+          setInputValue(newInputValue || "");
+
+        default:
+          break;
+      }
+    },
+  });
+
+  return (
+    <>
+      <div className="flex flex-col">
+        <Label {...getLabelProps()}>Pick some books:</Label>
+        <div
+          aria-label="menu-toggle"
+          {...getToggleButtonProps(getDropdownProps({ preventKeyAction: isOpen }))}
+          className="flex h-9 space-x-1 rounded-md border border-gray-300 p-1 focus-within:border-gray-900 hover:cursor-pointer hover:border-gray-400 focus-visible:border-gray-900">
+          {!props.renderNode
```

Comment:
I find the name 'renderNode' a bit confusing. Perhaps we can come up with a name that better explains what type of node it renders?

Reply:
Yeah 100%